libs/clustermatch/corr.py

```diff
@@ -0,0 +1,50 @@
+"""
+Functions to compute different correlation coefficients.
+
+All correlation functions in this module are expected to have the same input and output
+structure:
+
+ * The input is a pandas DataFrame with genes in rows (Ensembl IDs) and samples
+   columns. The values are gene expression data normalized with some technique,
+   but that should not be relevant for the correlation method.
+
+ * The output is a pandas DataFrame, a symmetric correlation matrix with genes
+   in rows and columns (Ensembl IDs), and the values are the correlation
+   coefficients. Diagonal values are expected to be ones.
+"""
+import pandas as pd
+import numpy as np
+from sklearn.metrics import pairwise_distances
+
+
+def pearson(data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Compute the Pearson correlation coefficient.
+    """
+    corr_mat = 1 - pairwise_distances(data.to_numpy(), metric="correlation", n_jobs=1)
```

Comment:
I'm bad at stats, why do you have to subtract the correlation from one? Is it because `pairwise_distances` wants to return something like a distance so a bigger number should be less similar?

Reply:
Yes, exactly. `pairwise_distances` always returns a distance, the correlation distance in this case (`1 - r`), but I want `r`, so I just do `1 - (1 - r) = r`. I was looking for a fast enough implementation of correlation coefficients, and this one seems pretty decent.